### PR TITLE
Fill PDMP Med Columns

### PR DESCRIPTION
**Pull request by mldorr**
This needs to be checked to see if it works with a patient with data in the quantity column in their EHR data. 
Note - I forgot to mention it in the commit message, but I also made the columns sort-able.

[Launch URL](https://launch.smarthealthit.org/launcher?launch_uri=https%3A%2F%2Fdeploy-preview-56--distracted-jepsen-bc9c39.netlify.com%2Flaunch.html&patient=5c41cecf-cf81-434f-9da7-e24e5a99dbc2&fhir_ver=4)

### CODE REVIEW

File: src/components/Summary.js [hunk 1]
```diff
@@ -344,6 +344,44 @@ export default class Summary extends Component {
     return <div className="graph-placeholder"></div>;
   }
 
+  //New section to modify alert language
+  mapAlertContent(content) {
```
Comment (achen2401) [inline src/components/Summary.js:348]: Maggie, are these changes intended?  I think we moved the logic to display alert text to summary.json file previously

File: src/components/summary.json [hunk 2]
```diff
@@ -137,11 +137,11 @@
            {
              "headers": {
                "Drug Description": "Name",
-               "Quantity": { "key": "Quantity", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": false},
+               "Quantity": { "key": "Quantity", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": true},
                "Written Date": { "key": "Start", "formatter": "datishFormat" },
                "Dispensed": { "key": "End", "formatter": "datishFormat" },
-               "Prescriber": { "key": "Prescriber", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": false},
-               "Pharmacy": { "key": "Pharmacy", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": false}
+               "Prescriber": { "key": "Prescriber", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": true},
```
Comment (achen2401) [inline src/components/summary.json:143]: I **think** you won't need the formatter here anymore, i.e. `{ "key": "Prescriber", "sortable": true}` the formatter function, `stringSubstitutionFormat`, was used previously to display empty string since we don't have the field yet.

File: src/components/Summary.js (hunk 1 above)
Comment (mldorr) [reply src/components/Summary.js:348]: Oof. No. I saw that in there and thought they just never got removed. I figured the pulls I did since that change was made would have updated the code on my computer but apparently not.

File: src/components/summary.json (hunk 2 above)
Comment (mldorr) [reply src/components/summary.json:143]: I didn't know what that did. Thanks!

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql [hunk 3]
```diff
@@ -811,6 +811,8 @@ define ReportOpioidMedicationRequests:
     Name:  ConceptText(O.medication as FHIR.CodeableConcept),
     Start: DateTimeText(O.authoredOn),
     End:   null as String, // MedicationRequest end date not supported in R4,
+    //Amount: QuantityText(O.value as FHIR.Quantity),
+    Quantity: O.dispenseRequest.quantity,
```
Comment (achen2401) [inline src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:815]: is it perhaps `O.dispenseRequest.quantity.value`? I was looking at the FHIR json `  "dispenseRequest": {
          "extension": [
            {
              "url": "http://cosri.org/fhir/pharmacy_name",
              "valueString": "REAL CHAIN STORE 1 (CO)"
            },
            {
              "url": "http://cosri.org/fhir/last_fill",
              "valueDate": "2020-03-28"
            }
          ],
          "quantity": {
            "value": 30
          },
          "expectedSupplyDuration": {
            "value": 8,
            "unit": "days",
            "system": "http://unitsofmeasure.org",
            "code": "d"
          }
        }`

Comment (mldorr) [reply src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:815]: I feel like I tried that first and got a processing error, but I can try it again. It does seem more correct.

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql [hunk 4]
```diff
@@ -851,6 +855,8 @@ define ReportNonOpioidMedicationRequests:
...
```
Comment (achen2401) [inline src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:859]: same as my previous comment, maybe it is `O.dispenseRequest.quantity.value`, based on the FHIR json: `  "dispenseRequest": {
          "extension": [
            {
              "url": "http://cosri.org/fhir/pharmacy_name",
              "valueString": "REAL CHAIN STORE 1 (CO)"
            },
            {
              "url": "http://cosri.org/fhir/last_fill",
              "valueDate": "2020-03-28"
            }
          ],
          "quantity": {
            "value": 30
          },
          "expectedSupplyDuration": {
            "value": 8,
            "unit": "days",
            "system": "http://unitsofmeasure.org",
            "code": "d"
          }
        }`

File: src/components/summary.json [hunk 5]
```diff
@@ -174,11 +174,11 @@
            {
              "headers": {
                "Drug Description": "Name",
-               "Quantity": { "key": "Quantity", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": false},
+               "Quantity": { "key": "Quantity", "formatter": "stringSubstitutionFormat", "formatterArguments": [""], "sortable": true},
```
Comment (achen2401) [inline src/components/summary.json:177]: same here, if we have the field information, we **probably** don't need `stringSubstitutionFormat` formatter here:
`{ "key": "Quantity", "sortable": true}`

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.json [hunk 6]
```diff
@@ -5093,6 +5278,17 @@
...
+                        "value" : {
+                           "path" : "quantity",
+                           "type" : "Property",
+                           "source" : {
+                              "path" : "dispenseRequest",
+                              "scope" : "O",
+                              "type" : "Property"
+                           }
+                        }
                      }, {
```
Comment (achen2401) [inline src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.json:5292]: is it possible to add the additional field (e.g. quantity) to the Test patient json:
https://github.com/uwcirg/AHRQ-CDS-Connect-PAIN-MANAGEMENT-SUMMARY/blob/master/src/utils/r4_test_patients/brenda_jackson.json#L757
so maybe we can test out the changes via smarthealthit launcher in dev instance...Just a thought about testing these changes..

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql (hunk 3 above)
Comment (achen2401) [reply src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:815]: @mldorr Maggie, I tried this out in my dev instance, I think this is correct:  **`Quantity: O.dispenseRequest.quantity.value.value`** note the additional, `.value`
Screen Shot 2020-08-11 at 9 40 12 AM

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql (hunk 4 above)
Comment (achen2401) [reply src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:859]: @mldorr Maggie, I think as I mentioned, this is correct: **`Quantity: O.dispenseRequest.quantity.value.value`** I re-build cql and the resulting json yields:
Screen Shot 2020-08-11 at 9 44 54 AM

File: src/components/summary.json (hunk 5 above)
Comment (achen2401) [reply src/components/summary.json:177]: noting here that I did need to change it to this:  `"Quantity": { "key": "Quantity", "sortable": true}` for the quantity to show

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.json (hunk 6 above)
Comment (achen2401) [reply src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.json:5292]: I did post some dummy MeddicationRequest data for brenda_jason (note, in **R4**) to test out

**Comment by achen2401** [comment]
@mldorr Maggie, I did just push my changes that will fill in the Quantity, Dispensed and Pharmacy columns for a **R4** patient (e.g. Brenda Jackson). CQL code changes are based on the JSON returned here:  https://fhir.cosri-demo.cirg.washington.edu/hapi-fhir-jpaserver/fhir/MedicationRequest?_format=json&patient=1
To test out you will need to do a git pull to get my latest changes and make sure, if you are testing in smarthealthit lancher, to test in R4 (i.e. iss = `http://launch.smarthealthit.org/v/r4/sim/eyJoIjoiMSIsImoiOiIxIn0/fhir` and/or fhirServiceUrl = `http://launch.smarthealthit.org/v/r4/sim/eyJoIjoiMSIsImoiOiIxIn0/fhir` in your public/launch-context.json file.  Also note the updated test data I added to Brenda Jackson so I can test.

**Comment by mldorr** [comment]
All looks good to me. @ivan-c @achen2401 I'd say go ahead and pull+merge.

File: src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql [hunk 7]
```diff
@@ -842,15 +850,20 @@ define NonOpioidMedicationStatements:
   C3F.ActiveOrCompletedMedicationStatement(C3F.MedicationStatementLookBack(
     [MedicationStatement: "Non opiod pain medications"],
     100 years //changed from 2 to 100
-  ))
+  ))
```
Comment (ivan-c) [inline src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql:853]:
```suggestion
  ))
```

**Comment by ivan-c** [comment]
> All looks good to me. @ivan-c @achen2401 I'd say go ahead and pull+merge.

@mldorr 

In the future, please mark as **Ready for review** if ready to merge, otherwise only admins can merge
